File: src/impl-go/RomanToInt.go
package main

import "fmt"

func RomanToInt() {
	// s := "LVIII"
	s := "MCMXCIV"
	result := 0

	fmt.Println("Calc me daddy")

	inputs := []rune(s)

	for i := 0; i < len(inputs); i++ {
		//fmt.Printf("char: %v\n", inputs[i])
		currentVal := lookupRomanValue(inputs[i])
		fmt.Printf("int: %v\n", currentVal)

		nextVal := 0

		if i+1 < len(inputs) {
			nextVal = lookupRomanValue(inputs[i+1])
		}

		if nextVal > currentVal {
			result = result + (nextVal - currentVal)
			i++
		} else {
			result += currentVal
		}
	}

	fmt.Printf("Input: %v : Output: %v\n", s, result)
}

func lookupRomanValue(val rune) int {
	switch val {
	case 'I':
		return 1
	case 'V':
		return 5
	case 'X':
		return 10
	case 'L':
		return 50
	case 'C':
		return 100
	case 'D':
		return 500
	case 'M':
		return 1000
	default:
		return 0
	}
}

File: src/impl-dotnet/RomanToInt.cs
namespace impl_dotnet
{
    public static class RomanToInt
    {
        public static KeyVal[] KeyVals = {
            new KeyVal {Key = 'I', Value = 1},
            new KeyVal {Key = 'V', Value = 5},
            new KeyVal {Key = 'X', Value = 10},
            new KeyVal {Key = 'L', Value = 50},
            new KeyVal {Key = 'C', Value = 100},
            new KeyVal {Key = 'D', Value = 500},
            new KeyVal {Key = 'M', Value = 1000}
        };

        public static void Calc()
        {
            Console.WriteLine("Calc me daddy");


            // string a = "LVIII";
            string s = "MCMXCIV";
            char[] inputs = s.ToArray();

            int result = 0;

            for (int i = 0; i < inputs.Length; i++)
            {
                char currentInput = inputs[i];
                char nextInput = '\0';

                if (i + 1 < inputs.Length)
                {
                    nextInput = inputs[i + 1];
                }

                int currentVal = 0;
                int nextVal = 0;

                foreach (KeyVal item in KeyVals)
                {
                    if (item.Key == currentInput)
                    {
                        currentVal = item.Value;
                    }

                    if (item.Key == nextInput)
                    {
                        nextVal = item.Value;
                    }
                }

                if (currentVal < nextVal)
                {
                    result = result + (nextVal - currentVal);
                    i++;
                }
                else
                {
                    result = result + currentVal;
                }

            }

            // for (int i = 0; i < KeyVals.Length; i++)
            // {
            //     KeyVal item = KeyVals[i];
            //     if(item.Key == a) {
            //         result += result + item.Value;
            //     }
            // }

            Console.WriteLine($"Input: {s} : Output: {result}");
        }
    }

    public struct KeyVal
    {
        public char Key { get; set; }
        public int Value { get; set; }
    }

}
